src/main/java/com/gw/web/GeoweaverController.java

```diff
@@ -924,13 +908,27 @@ public ResponseEntity<Resource> fileGetter(ModelMap model, @PathVariable(value="
 			String id = request.getParameter("id");
 			String mode = request.getParameter("mode");
 			String token = request.getParameter("token");
-			String history_id = BaseTool.isNull(request.getParameter("history_id")) ? 
+			String history_id = BaseTool.isNull(request.getParameter("history_id")) ?
 					new RandomString(18).nextString() : request.getParameter("history_id");
 			String[] hosts = request.getParameterValues("hosts[]");
 			String[] encrypted_password = request.getParameterValues("passwords[]");
 			String[] environments = request.getParameterValues("envs[]");
 			String[] passwords = RSAEncryptTool.getPasswords(encrypted_password, session.getId());
 			resp = wt.execute(history_id, id, mode, hosts, passwords, environments, token);
+
+            assert id != null;
```

Comment:
what will happen if id is null? why `wt.execute` won't fail if `id` is null here?

Reply:
it will throw a error if id is null, added it for testing. removing.